nexusclient/client.go

```diff
@@ -0,0 +1,725 @@
+// for the result until the provided context deadline is exceeded. When the deadline exceeds, GetResult will return a
+// nil response and [context.DeadlineExceeded] error. The client may issue multiple requests until the deadline exceeds
+// with a max request timeout of [nexusclient.Options.GetResultMaxRequestTimeout].
+func (h *OperationHandle) GetResult(ctx context.Context, options GetResultOptions) (*http.Response, error) {
+	switch h.state {
+	case nexusapi.OperationStateCanceled, nexusapi.OperationStateFailed:
+		return nil, &nexusapi.UnsuccessfulOperationError{State: h.state, Failure: h.failure}
+	case nexusapi.OperationStateSucceeded:
+		return h.response, nil
+	default:
+		if h.id == "" {
+			// This should never be possible as handles are constructed only by the client.
+			panic("handle has no id")
+		}
+		var unsuccessfulError *nexusapi.UnsuccessfulOperationError
+		response, err := h.client.getOperationResult(ctx, getOperationResultRequest{
+			Operation:   h.operation,
+			OperationID: h.id,
+			Header:      options.Header,
+			Wait:        options.Wait,
+		})
+		if err != nil {
+			if errors.As(err, &unsuccessfulError) {
+				h.state = unsuccessfulError.State
+				h.failure = unsuccessfulError.Failure
+			}
+		} else if response != nil {
+			h.response = response
+			h.state = nexusapi.OperationStateSucceeded
+		} else {
+			h.state = nexusapi.OperationStateRunning
+		}
+		return response, err
+	}
+}
+
+func (c *Client) getOperationResult(ctx context.Context, request getOperationResultRequest) (*http.Response, error) {
+	url, err := c.joinURL(request.Operation, request.OperationID, "result")
+	if err != nil {
+		return nil, err
+	}
+	httpReq, err := http.NewRequestWithContext(ctx, "GET", url.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	if request.Header != nil {
+		httpReq.Header = request.Header.Clone()
+	}
+	httpReq.Header.Set(headerUserAgent, UserAgent)
+
+	for {
+		response, err := c.sendGetOperationResultRequest(ctx, httpReq, request.Wait)
+		if err != nil {
+			if errors.Is(err, errOperationStillRunning) {
+				if request.Wait {
+					continue
+				} else {
+					return nil, nil
+				}
+			}
+			return nil, err
+		}
+		return response, nil
+	}
+}
+
+func (c *Client) sendGetOperationResultRequest(ctx context.Context, httpReq *http.Request, wait bool) (*http.Response, error) {
+	if wait {
+		url := httpReq.URL
+		timeout := c.Options.GetResultMaxRequestTimeout
+		if deadline, set := ctx.Deadline(); set {
+			timeout = time.Until(deadline)
+			if timeout > c.Options.GetResultMaxRequestTimeout {
+				timeout = c.Options.GetResultMaxRequestTimeout
+			}
+		}
+
+		q := url.Query()
+		q.Set(nexusapi.QueryWait, fmt.Sprintf("%dms", timeout.Milliseconds()))
+		url.RawQuery = q.Encode()
+	}
+
+	response, err := c.Options.HTTPClient.Do(httpReq)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.StatusCode == 200 {
```

Comment:
```suggestion
	if response.StatusCode == http.StatusOK {
```

Use constants in the HTTP package

Reply:
This one slipped, thanks.

Reply:
Hmm.. seems like it slipped in this entire package, I did it correctly in the server code.

Reply:
Yeah, trying not to repeat myself when I see something multiple times